### PR TITLE
A large change from cached lists to signals.

### PR DESCRIPTION
**Pull request by doctormo**

### CODE REVIEW

**Comment by SpaceK33z** [comment]
Hi, thanks for your PR. I can't merge it with the change in the license though, is this really necessary?

**Comment by doctormo** [comment]
I was wondering why you didn't mention that in the issue. The ISC license is pretty bad, but if you're interested in the code, I can request an exemption from the boss.

**Comment by SpaceK33z** [comment]
Oh, I didn't mention it because I assumed you'd change it. Why is ISC bad? It's the same as MIT, but with all redundant text removed ([source](https://en.wikipedia.org/wiki/ISC_license)).

**Comment by doctormo** [comment]
We're a copyleft engineering shop over here, so we tend to default to GPL where possible. I've got the nod to keep it ISC since it's not critical software for us.

**Comment by doctormo** [comment]
(I should say that I didn't answer why we think ISC bad: we think all MIT based licenses are pretty bad. Based on an equitable power relationship which doesn't exist between developers and large companies. Thus it's idealist and not as responsive as the GPL)

**Review by SpaceK33z — CHANGES_REQUESTED** [review]
Allright, thanks for changing it :). There are some comments I have about the current state:

- The tests seem to be failing, could you look into that? You can run the tests with `npm test` or take a look at the [CI output](https://travis-ci.org/CodeYellowBV/chartist-plugin-legend/builds/228838191#L545).
- Can you revert all current changes in the README?
- If you want you can add your name (and/or company) in the README, and in the `package.json` as `"contributors"` property.

File: chartist-plugin-legend.js [hunk 1]
```diff
@@ -140,74 +92,41 @@
                   options.position.insertBefore(legendElement, null);
                }
             });
+            chart.on('data', function (data) {
+              // empty when ready.
+              $(legendElement).empty();
```
Comment (SpaceK33z) [inline chartist-plugin-legend.js:97]: You're introducing a jQuery dependency here, could you use another way (I think `legendElement.innerHTML = ''` might work)?

**Comment by doctormo** [comment]
Ah I see. Sorry for the noise. We're not able to use npm or the testing framework at this time, which is a shame. You may not be able to merge this properly (it was a long shot). Thanks for trying with me.

**Pull request closed without merging**